packages/x-ui-guardian-bundle/src/react/smarts/GuardianSmart.ts

```diff
@@ -177,7 +191,9 @@ export class GuardianSmart<
         fetchPolicy: "network-only",
       })
       .then((response) => {
-        return response.data.me;
+        const user = response.data.me;
+
+        return this.transformUser(user);
```

Comment:
Make it async, what if user transformation can be done in such a way that we also dispatch event GuardianUserRetrievedEvent giving us ability to nicely extend it from other bundles?

Reply:
On second thought do something like GuardianUserTransformedEvent({ user }) inside transformUser.